dht/byte_ordered_partitioner.cc

```diff
@@ -147,17 +147,22 @@ token byte_ordered_partitioner::midpoint(const token& t1, const token& t2) const
 
 unsigned
 byte_ordered_partitioner::shard_of(const token& t) const {
+    return shard_of(t, _shard_count, 0);
```

Comment:
If I were the compiler, I wouldn't make a virtual call, because I know exactly which `shard_of` I want to call here. But I'm not sure.
@avikivity do you know?

Reply:
Unless `byte_ordered_partitioner` is `final`, nothing guarantees that there isn't a more derived class defined in some other TU overriding `shard_of()`.

Reply:
Then I shall make the class final.

Reply:
Note that making it `final` does not guarantee a devirtualization of the call, that depends on the compiler implementation.

Reply:
I devirtualized it, now both methods call a function instead of one method calling the other method.